src/Engine/Events/EventIR.cpp

```diff
@@ -838,300 +839,339 @@ std::string EventIR::toString() const {
     return fmt::format("{}: UNPROCESSED/{}", step, ::toString(type));
 }
 
-EventIR EventIR::parse(const RawEvent *evt, size_t size) {
+EventIR EventIR::parse(const RawEvent *evt, const size_t size) {
     EventIR ir;
     ir.type = evt->_e_type;
     ir.step = evt->v3;
 
-    auto requireSize = [&](size_t minSize) {
+    const auto end = &evt->_e_size + size;
+
+    bool requireSizeCalled = false;
+
+    // moving pointer to consume record details
+    auto remainingDetails = evt->details;
```

Comment:
We are generally agains the almost always auto, and spell the types out explicitly whenever it makes sense (e.g. as long as it's not `typename std::vector<std::pair<int, MyWhatever>>::const_iterator`).

Reply:
OK, makes sense, easier to review without checking out.